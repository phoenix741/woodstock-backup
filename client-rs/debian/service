[Unit]
Description=Woodstock Client Daemon

[Service]
ExecStart=/usr/bin/ws_client_daemon

[Install]
WantedBy=network.target