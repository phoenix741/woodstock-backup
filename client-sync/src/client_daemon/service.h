#pragma once

#include <QtService>

#include "woodstockclient.h"

class ClientService : public QtService::Service
{
    Q_OBJECT
public:
    explicit ClientService(int &argc, char **argv);

protected:
    virtual ClientService::CommandResult onStart() override;
    virtual ClientService::CommandResult onStop(int &exitCode) override;

private:
    std::unique_ptr<WoodstockClient> m_client;
};
