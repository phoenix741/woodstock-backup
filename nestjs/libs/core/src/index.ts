export * from './event-emitter';
export * from './commands';
export * from './config';
export * from './core.module';
export * from './input-output';
export * from './utils';
export * from './core.constant';
