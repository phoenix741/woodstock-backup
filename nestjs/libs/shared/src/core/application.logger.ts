import { Injectable, LoggerService } from '@nestjs/common';
import * as logform from 'logform';
import * as mkdirp from 'mkdirp';
import { join } from 'path';
import { createLogger, format, Logger, transports } from 'winston';
import 'winston-daily-rotate-file';

const { combine, timestamp, printf, colorize } = format;

const applicationFormat = printf((info: logform.TransformableInfo) => {
  return `${info.timestamp} [${(info.context || '').padEnd(25, ' ')}] ${info.level}: ${info.message}`;
});

@Injectable()
export class ApplicationLogger implements LoggerService {
  #logger: Logger;

  constructor(readonly worker?: string, readonly console = true) {
    const logPath = join(process.env.BACKUP_PATH || '', 'log');
    mkdirp(logPath);

    this.#logger = createLogger({
      level: process.env.LOG_LEVEL || 'info',
      format: combine(timestamp(), applicationFormat),
      transports: [
        ...(console
          ? [
              new transports.Console({
                format: combine(colorize(), timestamp(), applicationFormat),
              }),
            ]
          : []),
        new transports.DailyRotateFile({
          filename: join(logPath, `application-${worker}-%DATE%.log`),
          datePattern: 'YYYY-MM-DD',
          zippedArchive: true,
          maxSize: '2m', // Config
          maxFiles: '31d', // Config
          createSymlink: true,
          symlinkName: `application-${worker}.log`,
        }),
      ],
      exceptionHandlers: [
        new transports.Console({
          format: combine(colorize(), timestamp(), applicationFormat),
        }),
        new transports.DailyRotateFile({
          filename: join(logPath, `exceptions-${worker}-%DATE%.log`),
          datePattern: 'YYYY-MM-DD',
          zippedArchive: true,
          maxSize: '2m', // Config
          maxFiles: '31d', // Config
          createSymlink: true,
          symlinkName: `exceptions-${worker}.log`,
        }),
      ],
    });
  }

  log(message: string | Record<string, unknown>, context?: string): void {
    this.#logger.info(typeof message === 'string' ? { context, message } : { context, ...message });
  }

  error(message: string | Record<string, unknown>, trace?: string, context?: string): void {
    this.#logger.error(typeof message === 'string' ? { context, trace, message } : { context, trace, ...message });
  }

  warn(message: string | Record<string, unknown>, context?: string): void {
    this.#logger.warn(typeof message === 'string' ? { context, message } : { context, ...message });
  }

  debug(message: string | Record<string, unknown>, context?: string): void {
    this.#logger.debug(typeof message === 'string' ? { context, message } : { context, ...message });
  }

  verbose(message: string | Record<string, unknown>, context?: string): void {
    this.#logger.verbose(typeof message === 'string' ? { context, message } : { context, ...message });
  }
}
