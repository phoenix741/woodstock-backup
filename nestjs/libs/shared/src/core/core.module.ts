import { Module } from '@nestjs/common';
import { ConfigModule } from '@nestjs/config';
import { CommandsModule } from '../commands';
import { ApplicationConfigModule } from '../config';
import { FilesModule } from '../files';
import { InputOutputModule } from '../input-output';
import { ManifestModule } from '../manifest';
import { PoolModule } from '../pool';
import { RefcntModule } from '../refcnt';
import { ScannerModule } from '../scanner';
import { StatisticsModule } from '../statistics';

export const MODULES = [
  ApplicationConfigModule,
  CommandsModule,
  ConfigModule,
  FilesModule,
  InputOutputModule,
  ManifestModule,
  PoolModule,
  RefcntModule,
  ScannerModule,
  StatisticsModule,
];

@Module({
  imports: [...MODULES],
  providers: [],
  exports: [...MODULES],
})
export class CoreModule {}
