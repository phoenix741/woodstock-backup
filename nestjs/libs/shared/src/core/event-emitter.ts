import * as EventEmitter from 'events';
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
EventEmitter.defaultMaxListeners = 20;
