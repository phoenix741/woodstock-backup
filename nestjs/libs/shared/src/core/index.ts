export * from './event-emitter';
export * from './core.module';
export * from './application.logger';
