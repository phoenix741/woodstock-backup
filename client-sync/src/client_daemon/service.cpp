#include "service.h"

ClientService::ClientService(int &argc, char **argv) : Service{argc, argv}, m_client(new WoodstockClient())
{
}

ClientService::CommandResult ClientService::onStart()
{
    qDebug() << "Service was started";

    m_client->listen();

    qDebug() << "Service is listening";
    return ClientService::CommandResult::Completed; //service is now assumed started
}

ClientService::CommandResult ClientService::onStop(int &exitCode)
{
    qDebug() << "Stop received";
    
    exitCode = 0;

    m_client->stop();

    qDebug() << "Stopped";
    return ClientService::CommandResult::Completed;
}
