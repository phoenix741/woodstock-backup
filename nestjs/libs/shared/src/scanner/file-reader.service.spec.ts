import { Test, TestingModule } from '@nestjs/testing';
import { globStringToRegex } from '@woodstock/core';
import { AsyncIterableX, from, toArray } from 'ix/asynciterable';
import * as Long from 'long';
import { IndexManifest } from '../manifest/index-manifest.model.js';
import { FileManifest } from '../protobuf/woodstock.interface.js';
import { FileBrowserService } from './file-browser.service.js';
import { FileReaderService } from './file-reader.service.js';

describe('FileReader', () => {
  let service: FileReaderService;

  const fakeFileBrowserService = {
    getFiles() {
      return (): AsyncIterableX<FileManifest> => {
        return from([
          {
            path: Buffer.from('/file-reader.service.ts'),
            stats: {
              lastModified: Long.fromValue(122),
              size: Long.fromValue(123),
              mode: Long.fromValue(0o0100000),
            },
          } as FileManifest,
        ]);
      };
    },
  };

  beforeEach(async () => {
    const app: TestingModule = await Test.createTestingModule({
      providers: [FileReaderService, { provide: FileBrowserService, useValue: fakeFileBrowserService }],
    }).compile();

    service = app.get<FileReaderService>(FileReaderService);
  });

  test('Calculate hash of the files', async () => {
    const index = new IndexManifest();
    index.add({
      path: Buffer.from('/file-reader.service.ts'),
      stats: {
        lastModified: Long.fromValue(123),
        size: Long.fromValue(124),
        mode: Long.fromValue(0o0100000),
      },
      acl: [],
      xattr: {},
      chunks: [],
    });

    const value = await toArray(service.getFiles(index, Buffer.from(__dirname), [], [globStringToRegex('*.spec.ts')]));

    expect(value).toMatchSnapshot();
  });
});
