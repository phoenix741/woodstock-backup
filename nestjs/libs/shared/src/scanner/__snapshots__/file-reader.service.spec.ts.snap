// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileReader Calculate hash of the files 1`] = `
[
  {
    "manifest": {
      "chunks": [
        {
          "data": [
            178,
            235,
            139,
            144,
            22,
            249,
            60,
            125,
            39,
            86,
            83,
            226,
            122,
            244,
            28,
            231,
            70,
            27,
            239,
            132,
            194,
            30,
            161,
            204,
            47,
            108,
            132,
            182,
            217,
            15,
            12,
            25,
          ],
          "type": "Buffer",
        },
      ],
      "path": {
        "data": [
          47,
          102,
          105,
          108,
          101,
          45,
          114,
          101,
          97,
          100,
          101,
          114,
          46,
          115,
          101,
          114,
          118,
          105,
          99,
          101,
          46,
          116,
          115,
        ],
        "type": "Buffer",
      },
      "sha256": {
        "data": [
          178,
          235,
          139,
          144,
          22,
          249,
          60,
          125,
          39,
          86,
          83,
          226,
          122,
          244,
          28,
          231,
          70,
          27,
          239,
          132,
          194,
          30,
          161,
          204,
          47,
          108,
          132,
          182,
          217,
          15,
          12,
          25,
        ],
        "type": "Buffer",
      },
      "stats": {
        "lastModified": Long {
          "high": 0,
          "low": 122,
          "unsigned": false,
        },
        "mode": Long {
          "high": 0,
          "low": 32768,
          "unsigned": false,
        },
        "size": Long {
          "high": 0,
          "low": 123,
          "unsigned": false,
        },
      },
    },
    "type": 1,
  },
]
`;
